app/Http/Controllers/TransactionsController.php

```diff
@@ -0,0 +1,140 @@
+        return ['totalBalance' => $totalBalance, 'transactions' => $data];
+    }
+
+    public function create(Request $request)
+    {
+        /** @var User $user */
+        $user = $request->user();
+        //validation for inputs
+        $validator = Validator::make(
+            $request->all(),
+            [
+                'label' => 'required',
+                'amount' => 'required|numeric',
+                'date' => 'required|date'
+            ]
+        );
+        if ($validator->fails()) {
+            return $validator->errors();
+        }
+        $transaction = new Transaction();
+        $transaction->label = $request->label;
+        $transaction->amount = $request->amount;
+        $transaction->date = $request->date;
+        $user->transactions()->save($transaction);
+        return $transaction->id;
+    }
+
+    /**
+     * @param  Request  $request
+     * @param $id
+     * @return bool|string
+     */
+    public function update(Request $request, $id)
+    {
+        //validation for inputs
+        $validator = Validator::make(
+            $request->all(),
+            [
+                'label' => 'required',
+                'amount' => 'required|numeric',
+                'date' => 'required|date'
+            ]
+        );
+        if ($validator->fails()) {
+            return $validator->errors();
+        }
+        /** @var User $user */
+        $user = $request->user();
+        /** @var Transaction $transaction */
+        $transaction = $user->transactions()->findOrFail($id);
+        $transaction->label = $request->label;
+        $transaction->amount = $request->amount;
+        $transaction->date = $request->date;
+        return $transaction->save();
+    }
+
+    /**
+     * Deletes a transaction
+     * Ideally this would be limited to the user's transactions
+     * @param  Request  $request
+     * @param  int  $id
+     * @return bool
+     */
+    public function delete(Request $request, int $id): bool
+    {
+        /** @var User $user */
+        $user = $request->user();
+        try {
+            return $user->transactions()->findOrFail($id)->delete();
+        } catch (Exception $ex) {
+            return false;
+        }
+    }
+
+    public function import(Request $request)
+    {
+        /** @var User $user */
+        $user = $request->user();
+        $validator = Validator::make(
+            $request->all(),
+            [
+                'file' => 'required|mimes:csv,txt'
+            ]
+        );
+        if ($validator->fails()) {
+            // display errors on each form field
+            $errors = [];
+            foreach ($validator->errors()->getMessages() as $message) {
+                array_push($errors, $message[0]);
+            }
+            return ['message' => implode('. ', $errors)];
+        }
+
+        $file = $request->file('file');
+        $file->move(public_path(), $file->getClientOriginalName());
```

Comment:
The uploaded files will be publicly accessible. You should use `storage_path()` here.

Reply:
Resolved